app/forms/book_form.rb

```diff
@@ -0,0 +1,66 @@
+class BookForm < Rectify::Form
+  attribute :title, String
+  attribute :category_id, Integer
+  attribute :main_image, String
+  attribute :images, String
+  attribute :description, String
+  attribute :year, Integer
+  attribute :height, Integer
+  attribute :width, Integer
+  attribute :thickness, Integer
+  attribute :price, Decimal
+  attribute :author_ids, Array[Integer]
+  attribute :material_ids, Array[Integer]
+
+  REGEXP = /\A([\p{Alnum}!#$%&'*+-\/=?^_`{|}~\s])+\z/
+
+  validate :must_have_category, :must_have_authors, :must_have_materials
+
+  validates :title,
+            presence: true,
+            format: { with: REGEXP },
+            length: { maximum: 80 }
+  validates :description,
+            presence: true,
+            format: { with: REGEXP },
+            length: { maximum: 1000 }
+  validates :year, numericality: {
+    only_integer: true,
+    greater_than: 1990,
+    less_than_or_equal_to: Date.today.year
+  }
+  validates :height, numericality: {
+    greater_than: 7,
+    less_than: 16
+  }
+  validates :width, numericality: {
+    greater_than: 3,
+    less_than: 8
+  }
+  validates :thickness, numericality: {
+    greater_than: 0,
+    less_than: 4
+  }
+  validates :price, numericality: {
+    greater_than_or_equal_to: 0.50,
+    less_than_or_equal_to: 199.95
+  }
+
+  def must_have_category
+    errors.add(:base, tr('.empty_category')) if category_id.blank?
+  end
+
+  def must_have_authors
+    errors.add(:base, tr('.empty_authors')) if author_ids.all?(&:blank?)
```

Comment:
what do you want to check? maybe would be enough `author_ids.blank?`

Reply:
That`s because of the [Rails behavior](http://api.rubyonrails.org/classes/ActionView/Helpers/FormHelper.html#method-i-check_box) of treating checkboxes. As it states in the Gotcha section:

> To prevent this the helper generates an auxiliary hidden field before the very check box. The hidden field has the same name and its attributes mimic an unchecked check box.

So, the params always have an epmty string in `author_ids` array which allows to catch the situations when no authors were checked. Since
`[""].blank? # => false`
had to use this workaround
`[""].all?(&:blank?) # => true`

In closed Formtastic (used by Active Admin for creating entities` form components) issues [246](https://github.com/justinfrench/formtastic/issues/246) and [697](https://github.com/justinfrench/formtastic/issues/697) their authors insist to keep this default approach:

> This enables the controller to react to the fact that the user hasn't ticked any checkbox. If you leave out the hidden field, don't tick any checkbox and submit the form **the browser doesn't transmit anything for that attribute**. So you wouldn't be able to update/clear for example a has many association.

The same applies to `material_ids`.